core/state/trie_prefetcher_extra_test.go

```diff
@@ -0,0 +1,200 @@
+	}
+	got, err = levelDB.Get(countKey)
+	if err == nil {
+		count = binary.BigEndian.Uint64(got)
+	}
+	got, err = levelDB.Get(blockKey)
+	if err == nil {
+		block = binary.BigEndian.Uint64(got)
+	}
+
+	// Make a trie on the levelDB
+	address1 := common.Address{42}
+	address2 := common.Address{43}
+	addBlock := func(db Database, snaps *snapshot.Tree, kvsPerBlock int, prefetchers int) (statedb *StateDB) {
+		statedb, root, err = addKVs(db, snaps, address1, address2, root, block, kvsPerBlock, prefetchers)
+		require.NoError(err)
+		count += uint64(kvsPerBlock)
+		block++
+
+		return statedb
+	}
+
+	lastCommit := block
+	commit := func(levelDB ethdb.Database, snaps *snapshot.Tree, db Database) {
+		require.NoError(db.TrieDB().Commit(root, false))
+
+		for i := lastCommit + 1; i <= block; i++ {
+			require.NoError(snaps.Flatten(fakeHash(i)))
+		}
+		lastCommit = block
+
+		// update the tracking keys
+		err = levelDB.Put(rootKey, root.Bytes())
+		require.NoError(err)
+		err = database.PutUInt64(levelDB, blockKey, block)
+		require.NoError(err)
+		err = database.PutUInt64(levelDB, countKey, count)
+		require.NoError(err)
+	}
+
+	tdbConfig := &triedb.Config{
+		HashDB: &hashdb.Config{
+			CleanCacheSize: 3 * 1024 * 1024 * 1024,
+		},
+	}
+	db := NewDatabaseWithConfig(levelDB, tdbConfig)
+	snaps := snapshot.NewTestTree(levelDB, fakeHash(block), root)
+	for count < uint64(wantKVs) {
+		previous := root
+		_ = addBlock(db, snaps, 100_000, 0) // Note this updates root and count
+		b.Logf("Root: %v, kvs: %d, block: %d", root, count, block)
+
+		// Commit every 10 blocks or on the last iteration
+		if block%10 == 0 || count >= uint64(wantKVs) {
+			commit(levelDB, snaps, db)
+			b.Logf("Root: %v, kvs: %d, block: %d (committed)", root, count, block)
+		}
+		if previous != root {
+			require.NoError(db.TrieDB().Dereference(previous))
+		} else {
+			panic("root did not change")
+		}
+	}
+	require.NoError(levelDB.Close())
+	b.Logf("Starting benchmarks")
+	b.Logf("Root: %v, kvs: %d, block: %d", root, count, block)
+	for _, updates := range []int{100, 200, 500} {
+		for _, prefetchers := range []int{0, 1, 4, 16} {
+			b.Run(fmt.Sprintf("updates_%d_prefetchers_%d", updates, prefetchers), func(b *testing.B) {
+				startRoot, startBlock, startCount := root, block, count
+				defer func() { root, block, count = startRoot, startBlock, startCount }()
+
+				levelDB, err := rawdb.NewLevelDBDatabase(path.Join(dir, "level.db"), 0, 0, "", false)
+				require.NoError(err)
+				snaps := snapshot.NewTestTree(levelDB, fakeHash(block), root)
+				db := NewDatabaseWithConfig(levelDB, tdbConfig)
+				if prefetchers > 0 {
+					db = WithPrefetcher(db, prefetchers)
+				}
+				getMetric := func(metric string) int64 {
+					meter := metrics.GetOrRegisterMeter(triePrefetchMetricsPrefix+namespace+"/storage/"+metric, nil)
+					return meter.Snapshot().Count()
+				}
+				startLoads := getMetric("load")
+				for i := 0; i < b.N; i++ {
+					_ = addBlock(db, snaps, updates, prefetchers)
```

Comment:
You never use the returned value from `addBlock()` so it can be removed to clean up the `_ = ` in both places.